cocos/video/video-player.ts

```diff
@@ -374,14 +374,8 @@ export class VideoPlayer extends Component {
         } else {
             this._impl.syncClip(this._clip);
         }
-    }
+        this._cachedCurrentTime = 0;
```

Comment:
need to reset `_cachedCurrentTime` when change video source. Otherwise, new video source will play from old video source progress.